app.rb

```diff
@@ -1,8 +1,40 @@
-require 'sinatra/base'
+require "sinatra"
+require 'sinatra/reloader' if development?
+require 'pg'
 
 class Chitter < Sinatra::Base
-  get '/test' do
-    'Test page'
+
+  configure :development do
+    register Sinatra::Reloader
+  end
+
+  get "/" do
+    redirect :messages
+  end
+
+  get "/messages" do
+    @reversed = params[:reversed]
+    @find = params[:find]
+    connection = PG.connect(dbname: 'chitter_test', user: 'JMMakers', password: '1234')
+    result = connection.exec('SELECT * FROM peeps')
+    @results = result.map { |entry| entry }
+    if @find
+      find_results = []
+      @results.each { |entry| find_results << entry if entry["message"].downcase.include? @find.downcase }
```

Comment:
Perhaps searching the database using SQL commands could be quicker? Either way, this works quite well.